### ISSUE TITLE
Constants are not being considered when predicting fixed effects

### ISSUE THREAD

**Issue by libertango11** [sergiocorreia/reghdfe#140]
Hello,

I first want to thank you for making this great package. I think this is related to constant being displayed in the new versions. (I prefer the previous one though).

This might be a very minor issue, but:
For version 5.2.2, I've noticed that constants are not being taken account when predicting the fixed effects. For example:

, absorb(firm_id worker_id year_coefs=year_id) constant
, absorb(firm_id worker_id year_coefs=year_id) 

yielded different year_coefs in the previous versions (because of the constant). But for this version, they are generating same outputs even with "constant" options. Basically, I get fixed effects + constants for both of the commands.

Could you check this and make sure we get fixed effects coefficients without constant, when we have "constnat" in the options?

Thank you very much!
Peter

**Comment by sergiocorreia** [comment]
Just to be clear, which would be the desired  behavior? Say we have a constant, do you want it assigned to the fixed effects (as in the previous versions) or something else. Also, maybe the `noconstant` option is what you want? EG:

```stata
clear all
cls
sysuse auto
reghdfe price weight, a(A=turn)
reghdfe price weight, a(B=turn) constant
reghdfe price weight, a(C=turn) noconstant
su A B C
```

(Is the code above helpful?)

**Comment by libertango11** [comment]
Thanks! I examined a bit with 4.4.13 vs 5.2.6 and found the difference. I
am starting to think that this is an even more minor issue than I thought,
but..

clear all
sysuse auto

reghdfe price, a(A=i.rep78#i.foreign)
reghdfe price, a(B=i.rep78#i.foreign) constant
reghdfe price, a(C=i.rep78#i.foreign) noconstant

For 5.2.6version, A, B and C are all the same. But for 4.4.13, I get
different results for B. (For the first observation, AMC Concord, I get
461.0306 for 4.4.13-B. For others, I get 6607.0741)

I was wondering why only 4.4.13 would produce different output. Is there
any way to get 461.0306 using the latest version of reghdfe?

I really appreciate your help and feedback.
Peter

**Comment by sergiocorreia** [comment]
Hi Peter,

I spent some time digging into the code and indeed found and fixed a bug in 5.2.6

When regressions were run without any regressors besides the fixed effects, the first fixed effect still included the constant. Now, instead, the first fixed effect will only include the constant if the -noconstant- option was set.

**Comment by libertango11** [comment]
Thank you for your quick solution to the problem. I really appreciate it!

Peter
